libraries/chain/include/eosio/chain/types.hpp

```diff
@@ -99,25 +100,23 @@ namespace eosio::chain {
     * serialization and to/from variant
     */
    class shared_blob : public shared_string {
-      public:
-         shared_blob() = delete;
-         shared_blob(shared_blob&&) = default;
-
-         shared_blob(const shared_blob& s) = default;
-
+   public:
+      shared_blob() = default;
 
-         shared_blob& operator=(const shared_blob& s) = default;
+      shared_blob(shared_blob&&) = default;
+      shared_blob(const shared_blob& s) = default;
 
-         shared_blob& operator=(shared_blob&& ) = default;
+      explicit shared_blob(std::string_view s) : shared_string(s) {}
 
-         template <typename InputIterator>
-         shared_blob(InputIterator f, InputIterator l, const allocator_type& a)
-         :shared_string(f,l,a)
-         {}
+      template <typename InputIterator>
+      shared_blob(InputIterator f, InputIterator l) : shared_string(f, l) {}
 
-         shared_blob(const allocator_type& a)
-         :shared_string(a)
-         {}
+      shared_blob& operator=(const shared_blob& s) = default;
+      shared_blob& operator=(shared_blob&& ) = default;
+      shared_blob& operator=(std::string_view sv) {
+         static_cast<shared_string&>(*this) = sv;
+         return *this;
+      }
    };
```

Comment:
Seems like `shared_blob` should be move to `database_utils.hpp`. Maybe all the `shared_*` types in here should be moved to `database_utils.hpp`

Reply:
I don't mind moving them, but I think because these are type definitions, them being in `types.hpp` makes sense.

Reply:
I think having the typedefs in `types.hpp` is fine.